### ISSUE TITLE
Could we set "finetune_cnn_after" parameter to -1?

### ISSUE THREAD

**Issue by jessejchuang** [2017-fall-DL-training-program/ImageCaption#13]
Hi TA,

I don't know why the training speed is so slow in Azure VM and charge us so much money. Ran 5 epochs in the morning but still got no result in the evening.  **_The default value of "finetune_cnn_after" in opts.py is 0_**. Could we **_set it to -1_**? This can speed up our homework.

**Comment by connie980149** [comment]
I think it's okay to disable the fine tune function.
Can you report the time needed for one epoch?

**Comment by fansia** [comment]
It cost about **90 mins** per epoch for the first model.

**Comment by jessejchuang** [comment]
It seems the sample code doesn't summarize one epoch execution time. I can only try to estimate it by the time of one iteration. Check out the logs below. The time **'read data' and 'train'** in **an iteration** is around **1s vs. 0.5s**. So for **one epoch**( 113287 images, then 11329 iterations), that's **3h9m vs. 1h34m**.

- **Without "--finetune_cnn_after -1"**

('Read data:', _**0.13839316368103027**_)
iter 5997 (epoch 0), train_loss = 3.107, time/batch = **_0.780_**
('Read data:', **_0.14182162284851074_**)
iter 5998 (epoch 0), train_loss = 2.830, time/batch = **_0.788_**
('Read data:', **_0.2410578727722168_**)
iter 5999 (epoch 0), train_loss = 2.534, time/batch = **_0.801_**
image 184613: a group of people standing on a field with a horse
image 403013: a kitchen with a stove and a refrigerator


- **With "--finetune_cnn_after -1"**

('Read data:', **_0.012835979461669922_**)
iter 5997 (epoch 0), train_loss = 3.570, time/batch = **_0.453_**
('Read data:', **_0.01192784309387207_**)
iter 5998 (epoch 0), train_loss = 3.024, time/batch = **_0.448_**
('Read data:', **_0.011527299880981445_**)
iter 5999 (epoch 0), train_loss = 2.869, time/batch = **_0.446_**
image 184613: a group of people riding horses on a horse
image 403013: a kitchen with a kitchen counter top and a sink

**Comment by jessejchuang** [comment]
Forget it. The loss seems not to converge without cnn fine tuning.
